packages/hardhat-core/src/internal/util/io-ts.ts

```diff
@@ -1,5 +1,28 @@
 import * as t from "io-ts";
 
+export function optionalOrNullable<TypeT, OutputT, InputT>(
+  codec: t.Type<TypeT, OutputT, InputT>,
+  name: string = `${codec.name} | undefined`
+): t.Type<TypeT | undefined, OutputT | undefined, InputT | undefined | null> {
+  return new t.Type(
+    name,
+    (u: unknown): u is TypeT | undefined =>
+      u === undefined || u === null || codec.is(u),
+    (i, c) => {
+      if (i === undefined || i === null) {
+        return t.success(undefined);
+      }
+      return codec.validate(i, c);
+    },
+    (a) => {
+      if (a === undefined || a === null) {
```

Comment:
This `a === null` doesn't seem necessary either. Notice that the type of `a` is `TypeT | undefined`.